Fiber_Drawing_Tower/Arduino_and_Python_code/fiber_tower_control/fiber_tower_control.ino

```diff
@@ -0,0 +1,161 @@
+#include <AccelStepper.h>
+
+// setting all variables
+
+// flag to see if motor is already on
+bool cabestan_running = false;
+bool preform_motor_running = false;
+
+// Arduino pins for the drivers
+const int cabestan_stepPin = 4;
+const int cabestan_dirPin = 5;
+const int preform_stepPin = 2;
+const int preform_dirPin = 3;
+
+// Speed of motors
+int new_speed_cabestan;
+int new_speed_preform; 
+
+// Diameter variables
+float offset = 0.07;
+float diameter_tension;
+float conversion_factor_diameter_tension = 0.5;
+float real_diameter = 0;
+
+const int cabestan_max_speed = 999;
+const int preform_max_speed = 999;
+
+char command;
+int count;
+
+// input of the python app
+char motor_preform_dir;
+
+String received_diameter_string = "";
+float desired_diameter = 0;
+
+// Creates an instance for both motors
+AccelStepper cabestan_stepper(AccelStepper::DRIVER, cabestan_stepPin, cabestan_dirPin);
+AccelStepper preform_stepper(AccelStepper::DRIVER, preform_stepPin, preform_dirPin);
+
+// map function working with float 
+float mapf(float value, float fromLow, float fromHigh, float toLow, float toHigh) {
+  float result;
+  result = (value - fromLow) * (toHigh - toLow) / (fromHigh - fromLow) + toLow;
+  return result;
+} 
+
+// Function adjusting the cabestan motor speed with the value measured trought the potentiometer
+void controlCabestan(int cabestan_speed) {
+  cabestan_stepper.setSpeed(cabestan_speed);
+  cabestan_stepper.runSpeed();
+}
+
+// Function adjusting the preform motor speed and its direction depending the button pressed
+void controlPreformMotor(char dir, int preform_speed) {
+  if (dir == 't'){
+  preform_stepper.setSpeed(preform_speed);
+  preform_stepper.runSpeed();
+  }
+   if (dir == 'b'){
+  preform_stepper.setSpeed(-preform_speed);
+  preform_stepper.runSpeed();
+  }
+}
+
+void stepTheMotors() {
+  if (preform_motor_running) {
+    preform_stepper.runSpeed();
+  }
+
+  if (cabestan_running) {
+    cabestan_stepper.runSpeed();
+  }
+}
+
+void setup() {
+  // Initilization of serial port
+  Serial.begin(115200);
+  // Setting max speed of the motors to avoid damages
+  cabestan_stepper.setMaxSpeed(cabestan_max_speed);
+  preform_stepper.setMaxSpeed(preform_max_speed);
+}
+
+// Main loop
+void loop() {
+  if (Serial.available() > 0) {
+    command = Serial.read();
+
+    if (isPunct(command) || isDigit(command)) {
```

Comment:
maybe a comment here to explain what's going on

Reply:
Done